src/network/connection_dispatcher_task.cpp

```diff
@@ -0,0 +1,74 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// connection_dispatcher_task.cpp
+//
+// Identification: src/network/connection_dispatcher_task.cpp
+//
+// Copyright (c) 2015-2017, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "network/connection_dispatcher_task.h"
+
+#define MASTER_THREAD_ID (-1)
+
+namespace peloton {
+namespace network {
+
+ConnectionDispatcherTask::ConnectionDispatcherTask(int num_handlers,
+                                                   int listen_fd)
+    : NotifiableTask(MASTER_THREAD_ID), next_handler_(0) {
+  RegisterEvent(
+      listen_fd, EV_READ | EV_PERSIST,
+      METHOD_AS_CALLBACK(ConnectionDispatcherTask, DispatchConnection), this);
+  RegisterSignalEvent(SIGHUP, METHOD_AS_CALLBACK(NotifiableTask, ExitLoop),
+                      this);
+
+  // TODO(tianyu) Figure out what this initialization logic is doing and
+  // potentially rewrite
+  // register thread to epoch manager.
+  if (concurrency::EpochManagerFactory::GetEpochType() ==
+      EpochType::DECENTRALIZED_EPOCH) {
+    for (size_t task_id = 0; task_id < (size_t)num_handlers; task_id++) {
```

Comment:
I am not comfortable with the naming. It is clear that you use task_id here to conform with your task definition, But in the code is register thread.

Reply:
Will change

Reply:
Go with `task_id`